appservice/src/disconnectRepo.ts

```diff
@@ -6,16 +6,21 @@
 import { SiteSourceControl } from 'azure-arm-website/lib/models';
 import { MessageItem } from 'vscode';
 import { AzureTreeItem, IActionContext } from 'vscode-azureextensionui';
-import { editScmType, ISiteTreeRoot } from './';
+import { editScmType } from './';
 import { ext } from './extensionVariables';
 import { localize } from './localize';
 import { ScmType } from './ScmType';
+import { DeploymentsTreeItem } from './tree/DeploymentsTreeItem';
 
-export async function disconnectRepo(context: IActionContext, node: AzureTreeItem<ISiteTreeRoot>): Promise<void> {
-    const sourceControl: SiteSourceControl = await node.root.client.getSourceControl();
-    const disconnectButton: MessageItem = { title: localize('disconnect', 'Disconnect') };
-    const disconnect: string = localize('disconnectFromRepo', 'Disconnect from "{0}"? This will not affect your app\'s active deployment. You may reconnect a repository at any time.', sourceControl.repoUrl);
-    await ext.ui.showWarningMessage(disconnect, { modal: true }, disconnectButton);
-    await editScmType(node.root.client, node, context, ScmType.None);
-    await node.refresh();
+export async function disconnectRepo(context: IActionContext, node: DeploymentsTreeItem): Promise<void> {
```

Comment:
Yeah I don't think you actually need the node here. I would prefer if you had a call signature more similar to `editScmType`. Maybe even the exact same parameters just to be consistent. With all of these comments, I'm hoping you don't actually need to add `root` or `parent` to `DeploymentsTreeItem`